pkg/acquisition/modules/s3/s3.go

```diff
@@ -0,0 +1,594 @@
+	}
+
+	if sess == nil {
+		return fmt.Errorf("failed to create aws session")
+	}
+	config := aws.NewConfig()
+	if s.Config.AwsRegion != "" {
+		config = config.WithRegion(s.Config.AwsRegion)
+	}
+	if s.Config.AwsEndpoint != "" {
+		config = config.WithEndpoint(s.Config.AwsEndpoint)
+	}
+	s.sqsClient = sqs.New(sess, config)
+	if s.sqsClient == nil {
+		return fmt.Errorf("failed to create SQS client")
+	}
+	return nil
+}
+
+func (s *S3Source) readManager() {
+	logger := s.logger.WithField("method", "readManager")
+	for {
+		select {
+		case <-s.t.Dying():
+			logger.Infof("Shutting down S3 read manager")
+			s.cancel()
+			return
+		case s3Object := <-s.readerChan:
+			logger.Debugf("Reading file %s/%s", s3Object.Bucket, s3Object.Key)
+			err := s.readFile(s3Object.Bucket, s3Object.Key)
+			if err != nil {
+				logger.Errorf("Error while reading file: %s", err)
+			}
+		}
+	}
+}
+
+func (s *S3Source) getBucketContent() ([]*s3.Object, error) {
+	logger := s.logger.WithField("method", "getBucketContent")
+	logger.Debugf("Getting bucket content for %s", s.Config.BucketName)
+	bucketObjects := make([]*s3.Object, 0)
+	var continuationToken *string = nil
+	for {
+		out, err := s.s3Client.ListObjectsV2WithContext(s.ctx, &s3.ListObjectsV2Input{
+			Bucket:            aws.String(s.Config.BucketName),
+			Prefix:            aws.String(s.Config.Prefix),
+			ContinuationToken: continuationToken,
+		})
+		if err != nil {
+			logger.Errorf("Error while listing bucket content: %s", err)
+			return nil, err
+		}
+		bucketObjects = append(bucketObjects, out.Contents...)
+		if out.NextContinuationToken == nil {
+			break
+		}
+		continuationToken = out.NextContinuationToken
+	}
+	sort.Slice(bucketObjects, func(i, j int) bool {
+		return bucketObjects[i].LastModified.Before(*bucketObjects[j].LastModified)
+	})
+	return bucketObjects, nil
+}
+
+func (s *S3Source) listPoll() error {
+	logger := s.logger.WithField("method", "listPoll")
+	ticker := time.NewTicker(time.Duration(s.Config.PollingInterval) * time.Second)
+	lastObjectDate := time.Now()
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-s.t.Dying():
+			logger.Infof("Shutting down list poller")
+			s.cancel()
+			return nil
+		case <-ticker.C:
+			newObject := false
+			bucketObjects, err := s.getBucketContent()
+			if err != nil {
+				logger.Errorf("Error while getting bucket content: %s", err)
+				continue
+			}
+			if bucketObjects == nil {
+				continue
+			}
+			for i := len(bucketObjects) - 1; i >= 0; i-- {
+				if bucketObjects[i].LastModified.After(lastObjectDate) {
+					newObject = true
+					logger.Debugf("Found new object %s", *bucketObjects[i].Key)
+					s.readerChan <- S3Object{
+						Bucket: s.Config.BucketName,
+						Key:    *bucketObjects[i].Key,
+					}
+				} else {
+					break
+				}
+			}
+			if newObject {
+				lastObjectDate = *bucketObjects[len(bucketObjects)-1].LastModified
+			}
+		}
+	}
+}
+
+func (s *S3Source) extractBucketAndPrefix(message *string) (string, string, error) {
```

Comment:
Here it's better to check the type of message and set it to avoid checking each time the message type.